src/sentry/web/frontend/setup_wizard.py

```diff
@@ -90,22 +93,43 @@ def get(self, request: HttpRequest, wizard_hash) -> HttpResponse:
 
         # Fetching or creating a token
         token = None
-        tokens = [
-            x
-            for x in ApiToken.objects.filter(user=request.user).all()
-            if "project:releases" in x.get_scopes()
-        ]
-        if not tokens:
-            token = ApiToken.objects.create(
-                user=request.user,
-                scope_list=["project:releases"],
-                refresh_token=None,
-                expires_at=None,
+        serialized_token = None
+
+        can_use_org_tokens = len(orgs) == 1 and features.has(
+            "organizations:org-auth-tokens", orgs[0], actor=request.user
+        )
+
+        if can_use_org_tokens:
+            org = orgs[0]
+            token_str = generate_token(org.slug, generate_region_url())
+            token_hashed = hash_token(token_str)
+            token = OrgAuthToken.objects.create(
+                name="Sentry-Wizard-Generated-Token",
+                organization_id=org.id,
+                scope_list=["org:ci"],
+                created_by_id=request.user.id,
+                token_last_characters=token_str[-4:],
+                token_hashed=token_hashed,
             )
+            serialized_token = serialize(token, request.user, token=token_str)
         else:
-            token = tokens[0]
+            tokens = [
+                x
+                for x in ApiToken.objects.filter(user=request.user).all()
+                if "project:releases" in x.get_scopes()
+            ]
+            if not tokens:
+                token = ApiToken.objects.create(
+                    user=request.user,
+                    scope_list=["project:releases"],
+                    refresh_token=None,
+                    expires_at=None,
+                )
+            else:
+                token = tokens[0]
```

Comment:
```suggestion
	    tokens = ApiToken.objects.filter(user=request.user)
            token = next((token for token in tokens if "project:releases" in token.get_scopes()), None)
            
            if token is not None:
                token = ApiToken.objects.create(
                    user=request.user,
                    scope_list=["project:releases"],
                    refresh_token=None,
                    expires_at=None,
                )
```

Reply:
Much nicer! Glad to learn some Python again for a change 😅

Reply:
Oh yeah, my bad, good catch!